config.ini

```diff
@@ -1,181 +1,186 @@
-limit_to_mask: True
-bins: 20
-templates= ./templates/template1.png
-           ./templates/template2.png
-           ./templates/template3.png
-           ./templates/template4.png
-
-[HistogramModule.getHistogram]
-limit_to_mask: True
-bins: 20
-
-[BrightContrastModule.getContrast]
-limit_to_mask: True
-
-
-[BrightContrastModule.getBrightnessGray]
-limit_to_mask: True
-
-[BrightContrastModule.getBrightnessByChannelinColorSpace:RGB]
-limit_to_mask: True
-
-
-[BrightContrastModule.getBrightnessByChannelinColorSpace:YUV]
-limit_to_mask: True
-#pick a color space in the list from ‘RGB’, ‘HSV’, ‘RGB CIE’, ‘XYZ’, ‘YUV’, ‘YIQ’, ‘YPbPr’, ‘YCbCr’ : http://scikit-image.org/docs/dev/api/skimage.color.html#skimage.color.convert_colorspace
-to_color_space: YUV
-
-[SaveModule.saveFinalMask]
-overlay: True
-
-[SaveModule.saveThumbnails]
-image_work_size: 1.25x
-small_dim: 500
-
-[BlurDetectionModule.identifyBlurryRegions]
-image_work_size = 2.5x
-blur_radius: 7
-blur_threshold: .15
-
-
-[BasicModule.finalComputations]
-
-[BasicModule.finalProcessingSpur]
-disk_radius: 25
-
-[BasicModule.finalProcessingArea]
-area_thresh: 90000
-
-[DeconvolutionModule.seperateStains]
-;hed_from_rgb: Hematoxylin + Eosin + DAB
-;hdx_from_rgb: Hematoxylin + DAB
-;fgx_from_rgb: Feulgen + Light Green
-;bex_from_rgb: Giemsa stain : Methyl Blue + Eosin
-;rbd_from_rgb: FastRed + FastBlue + DAB
-;gdx_from_rgb: Methyl Green + DAB
-;hax_from_rgb: Hematoxylin + AEC
-;bro_from_rgb: Blue matrix Anilline Blue + Red matrix Azocarmine + Orange matrix Orange-G
-;bpx_from_rgb: Methyl Blue + Ponceau Fuchsin
-;ahx_from_rgb: Alcian Blue + Hematoxylin
-;hpx_from_rgb: Hematoxylin + PAS
-stain: hed_from_rgb
-use_mask: True
-
+[pipeline]
+
+steps= BasicModule.getBasicStats
+    ClassificationModule.byExampleWithFeatures:pen_markings
+    ClassificationModule.byExampleWithFeatures:coverslip_edg
+    AnnotationModule.maskOutAnnotation
+    LightDarkModule.getIntensityThresholdPercent:tissue
+#    BubbleRegionByRegion.roiWise
+    LightDarkModule.getIntensityThresholdPercent:darktissue
+    MorphologyModule.removeSmallObjects
+    MorphologyModule.fillSmallHoles
+    BlurDetectionModule.identifyBlurryRegions
+    BasicModule.finalProcessingSpur
+    BasicModule.finalProcessingArea
+    HistogramModule.compareToTemplates
+    HistogramModule.getHistogram
+    BrightContrastModule.getContrast
+    BrightContrastModule.getBrightnessGray
+    BrightContrastModule.getBrightnessByChannelinColorSpace:RGB
+    BrightContrastModule.getBrightnessByChannelinColorSpace:YUV
+    DeconvolutionModule.seperateStains
+    SaveModule.saveFinalMask
+    SaveModule.saveThumbnails
+    BasicModule.finalComputations
+
+	
+[BaseImage.BaseImage]
+image_work_size = 1.25x
+
+#three options: relative2mask, absolute, relative2image
+mask_statistics = relative2mask
+
+[BasicModule.getBasicStats]
+image_work_size = 1.25x
+
+[BasicModule.getMag]
+confirm: False
+
+[AnnotationModule.maskOutAnnotation]
+#annotation_path = []

```

Comment:
its okay to have this commented out if there is a reasonable default in your code, but i prefer to have it commented out *with* the default value stored here, so users can understand what the expected behavior is

Reply:
The default value is the directory that the images are in. Is there a way to communicate that in the config file?

Reply:
tricky. just put a note above the comment to explain, i don't there is a notation which will be obvious in this case. also mention the naming convention, which i assume is base_filename_image.svs - >base_filename_image.xml